kratos/tests/geometries/test_tetrahedra_3d_10.cpp

```diff
@@ -0,0 +1,169 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:
+//
+
+// System includes
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "geometries/tetrahedra_3d_10.h"
+#include "tests/geometries/test_geometry.h"
+#include "tests/geometries/test_shape_function_derivatives.h"
+#include "tests/geometries/cross_check_shape_functions_values.h"
+
+namespace Kratos {
+namespace Testing {
+
+  // /// Factory functions
+
+namespace{
+  Geometry<Node<3>>::PointsArrayType GenerateReferenceNodes()
+  {
+      Geometry<Node<3>>::PointsArrayType points;
+      points.push_back(Node<3>::Pointer(new Node<3>(1, 0.0, 0.0, 0.0)));
```

Comment:
The only thing I would recommend is to use a typedef for the Node 3, because we will remove some day the template argument (after removing it from the point is a meaningless argument).